Lab5/src/controllers/user/user.controller.ts

```diff
@@ -0,0 +1,90 @@
+import {UserService} from "../../services/services";
+import {Request, Response, NextFunction} from "express";
+import {validateOrReject} from "class-validator";
+import {plainToInstance} from "class-transformer";
+import {UserEntity} from "../../entity/user/user.entity";
+import {UserProperties} from "../../common/types/user/user.properties";
+import {UserValidator} from "../../entity/user/user-validator";
+
+class UserController {
+
+    private readonly userService: UserService;
+    private readonly userValidator: UserValidator;
+
+
+    constructor(userService: UserService, userValidator: UserValidator) {
+        this.userService = userService;
+        this.userValidator = userValidator;
+    }
+
+    get = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+        try {
+            const id: number = Number(req.params.id);
+            this.userValidator.isIdValid(id);
+            const user: UserEntity | undefined = await this.userService.get(id);
+            res.json(user);
+        } catch (e) {
+            next(e);
+        }
+    };
+
+    getAll = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+        try {
+            const users: UserEntity[] = await this.userService.getAll();
+            res.json(users);
+        } catch (e) {
+            next(e);
```

Comment:
what is a reason of that?

Reply:
Pass all exceptions to global validation middleware, instead of write every time res.status(XXX).end(XXX)

Reply:
then you don't need catch. Exception will be passed there. The question was why you use try catch if you actually ignore it